framework/wazuh/core/cluster/common.py

```diff
@@ -1794,7 +1795,7 @@ def as_wazuh_object(dct: Dict):
             return datetime.datetime.fromisoformat(dct['__wazuh_datetime__'])
         elif '__unhandled_exc__' in dct:
             exc_data = dct['__unhandled_exc__']
-            return eval(exc_data['__class__'])(*exc_data['__args__'])
+            return ast.literal_eval(exc_data['__class__'])(*exc_data['__args__'])
```

Comment:
I've been reviewing this to better understand it. I think we need to keep using `eval` instead of `literal_eval`, although it should not be a problem since the evaluated data is not a user input. 

This `dct` will contain an exception raised in any other part of the framework/cluster using the `WazuhJSONEncoder`, like in this example:
https://github.com/wazuh/wazuh/blob/89441d534a4915cee7bab6bb88d1af7d9dd934f4/framework/wazuh/core/cluster/worker.py#L523-L526

When the exception is known, like a Wazuh one, this information is included inside `WazuhJSONEncoder`:
https://github.com/wazuh/wazuh/blob/89441d534a4915cee7bab6bb88d1af7d9dd934f4/framework/wazuh/core/cluster/common.py#L1751-L1754

And when the exception is unknown/unhandled, the exception class name is included (for example, `ValueError` could be included as a string inside `__class__`):
https://github.com/wazuh/wazuh/blob/89441d534a4915cee7bab6bb88d1af7d9dd934f4/framework/wazuh/core/cluster/common.py#L1762-L1764

When an exception class name is evaluated using `eval` it is transformed into a real exception:
```PYTHON
>>> eval('ValueError')
<class 'ValueError'>
>>> 
>>> 
>>> eval('ValueError')('Test message')
ValueError('Test message')
>>> 
>>> raise eval('ValueError')('Test message')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Test message

```

But `ast.literal_eval` does not evaluates this kind of values:
```Python
>>> ast.literal_eval('ValueError')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/ast.py", line 110, in literal_eval
    return _convert(node_or_string)
  File "/usr/lib/python3.10/ast.py", line 109, in _convert
    return _convert_signed_num(node)
  File "/usr/lib/python3.10/ast.py", line 83, in _convert_signed_num
    return _convert_num(node)
  File "/usr/lib/python3.10/ast.py", line 74, in _convert_num
    _raise_malformed_node(node)
  File "/usr/lib/python3.10/ast.py", line 71, in _raise_malformed_node
    raise ValueError(msg + f': {node!r}')
ValueError: malformed node or string on line 1: <ast.Name object at 0x7f2dbaf7f430>

```

Reply:
Indeed, `ast.literal_eval` will only work for the exceptions encoded with `WazuhJSONEncoder` and not for unknown/unhandled ones. I agree we could keep using `eval` given that the input is generated by ours.

I will close the issue as not planned with this explanation and leave a note in the vulnerablity alert.